apps/heft/src/cli/HeftCommandLineUtilities.ts

```diff
@@ -249,60 +141,88 @@ export class HeftCommandLineUtilities {
   public registerStringListParameter(
     options: IRegisterParameterWithArgumentOptions
   ): IHeftStringListParameter {
-    const actionParameterMap: Map<CommandLineAction, CommandLineStringListParameter> = new Map();
-    for (const action of this._getActions(options.associatedActionNames)) {
+    return this._registerParameter(
+      options,
+      (action: CommandLineAction) => action.defineStringListParameter(options),
+      (parameter: CommandLineStringListParameter) => parameter.values,
+      (parameter: CommandLineStringListParameter) => (parameter.values || []).length > 0
+    );
+  }
+
+  private _registerParameter<
+    TTSCommandLineParameter extends CommandLineParameter,
+    TValue extends CustomActionParameterType
```

Comment:
Do we need to extend `CustomActionParameterType` or can we just leave it as `TAction`?